src/pages/index.js

```diff
@@ -150,13 +229,35 @@ function getCardElement(data) {
   const cardElementImage = cardElement.querySelector(".card__image");
   const likeButton = cardElement.querySelector(".card__like-button");
   const trashButton = cardElement.querySelector(".card__trash-button");
+  let selectedCard;
+  let selectedCardId;
+
+  data.isLiked ? likeButton.classList.add("card__like-button_liked") : null;
 
   likeButton.addEventListener("click", () => {
-    likeButton.classList.toggle("card__like-button_liked");
+    if (!likeButton.classList.contains("card__like-button_liked")) {
+      api
+        .addLike(data._id)
+        .then((res) => {
+          likeButton.classList.toggle("card__like-button_liked");
+        })
+        .catch(console.error);
+    } else {
+      api
+        .removeLike(data._id)
+        .then((res) => {
+          likeButton.classList.toggle("card__like-button_liked");
+        })
+        .catch(console.error);
+    }
   });
 
   trashButton.addEventListener("click", () => {
-    cardElement.remove();
+    handleDeleteCard(cardElement, data);
+  });
+
+  deleteCancelButton.addEventListener("click", () => {
```

Comment:
Please, move it onto line 310


Please, remove it from `getCardElement`

The `cancel` button should be handled only **once** in the file body, otherwise you add listeners again and again to the same elements. This may cause **`memory leak`**